src/js/angular/sparql-editor/controllers.js

```diff
@@ -289,5 +304,63 @@ function SparqlEditorCtrl($scope, $location, $jwtAuth, $repositories, toastr, $t
             });
     }
 
+    // ================================
+    // =     Setup output handlers    =
+    // ================================
+    outputHandlers.set('downloadAs', downloadAsHandler);
+
+    function downloadAsHandler(downloadAsEvent) {
+        const handler = downloadAsPluginNameToEventHandler.get(downloadAsEvent.pluginName);
+        if (handler) {
+            handler(downloadAsEvent);
+        }
+    }
+
+    // ================================
+    // =   Setup download handlers    =
+    // ================================
+
+    downloadAsPluginNameToEventHandler.set('response', downloadCurrentResults);
+    downloadAsPluginNameToEventHandler.set('extended_table', downloadThroughServer);
+
+    function downloadCurrentResults(downloadAsEvent) {
+        if ("application/sparql-results+json" === downloadAsEvent.contentType) {
+            ontoElement.getEmbeddedResultAsJson()
+                .then((response) => {
+                    const content = JSON.stringify(response, null, '\t');
+                    downloadAsFile(`${getFileTimePrefix()}_queryResults.json`, downloadAsEvent.contentType, content);
+                });
+        } else if ("text/csv" === downloadAsEvent.contentType) {
+            ontoElement.getEmbeddedResultAsCSV()
+                .then((response) => {
+                    downloadAsFile(`${getFileTimePrefix()}_queryResults.csv`, downloadAsEvent.contentType, response);
+                });
+        }
+    }
+
+    function getFileTimePrefix() {
+        const now = new Date();
+        return `${now.toLocaleDateString($scope.language)}_${now.toLocaleTimeString($scope.language)}`;
+    }
+
+    function downloadThroughServer(downloadAsEvent) {
+        const query = downloadAsEvent.query;
+        const infer = downloadAsEvent.infer;
+        const sameAs = downloadAsEvent.sameAs;
+        const accept = downloadAsEvent.contentType;
+        const authToken = localStorage.getItem('com.ontotext.graphdb.auth') || '';
+
+        // TODO change it
```

Comment:
Didn't we discuss that this should be changed to use formdata?! See here for example https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects